### PR TITLE
Changes in task request details page

### PR DESCRIPTION
**Pull request by Ajeyakrishna-k**
Date: 20 November 2023
Developer Name: @Ajeyakrishna-k 

----

## :bangbang:  Please merge the following PR first : 
- https://github.com/Real-Dev-Squad/website-backend/pull/1696
----

## Issue Ticket Number:- 
- #603 

## Description: 
- Uncomment the reject button and minor refactors in html.
- Adds event listener to reject button.
- Adds a spinner when there's an api call.
- Updates the status element in page when a status update CTA was called. 

Is Under Feature Flag 
- [x] Yes
- [ ] No

Database changes
- [ ] Yes
- [x] No

Breaking changes (If your feature is breaking/missing something please mention pending tickets)
- [ ] Yes
- [x] No

Is Development Tested?

- [x] Yes
- [ ] No

### Add relevant Screenshot below ( e.g test coverage etc. )

https://github.com/Real-Dev-Squad/website-dashboard/assets/98796547/12e48220-0506-4aac-8fa6-a7ffcfb2cee6

#### Device Responsiveness: 

https://github.com/Real-Dev-Squad/website-dashboard/assets/98796547/d0539dc7-d5cb-4ff9-8c29-9e60eac8f11c

Screenshot 2023-11-21 at 2 43 56 PM

### CODE REVIEW

File: taskRequests/details/script.js [hunk 1]
```diff
@@ -147,9 +159,12 @@ function getAvatar(user) {
   });
 }
 
-async function approveTaskRequest(userId) {
+async function updateTaskRequest(action, userId) {
+  const removeSpinner = addSpinner(container);
```
Comment (vinit717) [inline taskRequests/details/script.js:163]: here can't get by naming like what you are trying to do so can you explain

**Comment by vinit717** [comment]
Can you send the video which works for different screen sizes

File: taskRequests/details/script.js [hunk 2]
```diff
@@ -66,6 +68,16 @@ function renderTaskRequestDetails(taskRequest) {
   );
 }
 
+function updateStatus(status) {
+  const statusText = document.getElementById('taskRequest__status_text');
+  statusText.classList = [];
+  statusText.classList.add('taskRequest__status__chip');
+  statusText.classList.add(
+    `taskRequest__status__chip--${status?.toLowerCase()}`,
+  );
```
Comment (ajoykumardas12) [inline taskRequests/details/script.js:77]: Both classes can be added in single line as comma separated values.

Comment (Ajeyakrishna-k) [reply taskRequests/details/script.js:77]: Isn't this more readable?

Comment (ajoykumardas12) [reply taskRequests/details/script.js:77]: Single line would be easier to understand both classes are being added to same element.
But sure! Current one is more readable considering large class names.

**Comment by ajoykumardas12** [comment]
Does the reject button rejects a particular requester or all requests for that task?
It's not clear from the UI.

**Comment by Ajeyakrishna-k** [comment]
> Does the reject button rejects a particular requester or all requests for that task? It's not clear from the UI.

All users request. Design for this page will be improved.

**Comment by Ajeyakrishna-k** [comment]
> Can you send the video which works for different screen sizes

done..

**Review by ajoykumardas12 — APPROVED** [review]

**Review by vinit717 — APPROVED** [review]

**Comment by prakashchoudhary07** [comment]
Please resolve conflicts

**Comment by Ajeyakrishna-k** [comment]
> Please resolve conflicts

Done.

File: __tests__/taskRequests/taskRequest.test.js [hunk 3]
```diff
@@ -12,7 +12,7 @@ describe('Task Requests', () => {
 
   jest.setTimeout(60000);
 
-  beforeAll(async () => {
+  beforeEach(async () => {
```
Comment (vinit717) [inline __tests__/taskRequests/taskRequest.test.js:15]: why we are checking test before each individual test case?
why not beforeAll

**Review by vinit717 — APPROVED** [review]

**Pull request merged**